Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -1 +1,271 @@
 
+## Notes for Flutter TDD Clean Architecture Course Videos
+
+#### 1. Explanation & Project Structure
+> Video:https://www.youtube.com/watch?v=KjE2IDphA_U&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech
+
+In the `first tutorial` an explanation is given of the importance of having all functions tested in our application.  
+It also explains the **clean architecture**.
+All the features of the application will be divided into three folders, **domain**, **data** and **presentation**.
+It is also explained how each of these folders works and what they will contain.
+
+<br />
+
+#### 2. Entities & Use Cases 
+> Video:https://www.youtube.com/watch?v=lPkWX8xFthE&list=PLB6lc7nQ1n4iYGE_khpXRdJkJEp9WOech&index=2
+
+In the `second tutorial` we start by adding the packages of the dependencies that will be needed to build the application.
+
+````
+dependencies:
+  flutter:
+    sdk: flutter
+
+  get_it: ^2.0.1
+  
+  # Bloc for state management
+  flutter_bloc: ^0.21.0
+  
+  # Value equality
+  equatable: ^0.4.0
+  
+  # Functional programming thingies
+  dartz: ^0.8.6
+  
+  # Remote API
+  connectivity: ^0.4.3+7
+  
+  http: ^0.12.0+2
+  # Local cache
+  shared_preferences: ^0.5.3+4
+
+  cupertino_icons: ^0.1.2
+
+dev_dependencies:
+  flutter_test:
+    sdk: flutter
+  mockito: ^4.1.0
+  ````
+  
+  It is explained how the **entity** will work in this application.
+  The following explains the function of the **use cases** that will collect data from the repositories.
+  The use cases function in this application will collect the entities and show them in the presentation folder where all the elements that we will use physically in our application are located.
+  
+  
+  #### The either Type
+  
+  The **either** type is shown and explained with the function of being able to represent two types at the same time in this case serves for the errors and for the numbers that we will show.
```

Comment:
36 word sentence. 
```suggestion
  The **`either`** type means the function can represent two types. In this case it can be _either_ an **error** or a **number**.
```